src/OrleansConsulUtils/LegacyConsulMembershipConfigurator.cs

```diff
@@ -1,34 +1,15 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
-using Orleans;
-using Orleans.ConsulUtils.Configuration;
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans.Hosting;
 using Orleans.Runtime.Configuration;
-using Orleans.Runtime.Membership;
 
 namespace Orleans.Runtime.MembershipService
 {
     /// <inheritdoc cref="ILegacyMembershipConfigurator"/>
     public class LegacyConsulMembershipConfigurator : ILegacyMembershipConfigurator
     {
-        private readonly ILoggerFactory loggerFactory;
-        private readonly GlobalConfiguration globalConfiguration;
-        public LegacyConsulMembershipConfigurator(ILoggerFactory loggerFactory, GlobalConfiguration globalConfiguration)
+        public void ConfigureServices(GlobalConfiguration configuration, IServiceCollection services)
```

Comment:
as a design principle. we shouldn't put dependency in APIs. As in this API is not extensible at all. What if there's one more dependency, which we need to inject from DI anyways...

Reply:
Hmm, not sure I follow, this IS just a way to turn GlobalConfiguration into whatever registration we need in DI... it's not injecting general dependencies, it's just passing the only relevant piece of info. See that I removed logger or any other potential dependencies, this should just be a conversion from legacy config to configuring DI

Reply:
If you look from the lens of ILegacyMembershipConfigurator, gobalConfiguration and ServiceCollection is dependency for the configurator. 
I know it is not the case, here, but what if the configurator needs a third dependency from DI? Then this interface is not extensible.

Reply:
The point is that this doesn't run after the silo started. This runs when the host builder is configuring things. So it uses the same extension methods that a normal user would use for configuring the host. If it needs something from the container, it means that the user wouldn't be able to configure the same membership by not using the legacy converter.

Reply:
If it needs something from the container, it means that the user wouldn't be able to configure the same membership by not using the legacy converter.

I'm not sure what you mean by this. Our code behavior are very similar, we both configure membership when host builder is configuring things. 

it uses the same extension methods that a normal user would use for configuring the host.

Sure this is better. I will take a look on your second commit and see if we can merge this.